### PR TITLE
Make use of router-generator's getConfig function at router-vite-plugin

### PR DESCRIPTION
**Pull request by wuarmin**
Hey,
I noticed that the vite-plugin was not using `getConfig` of `router-generator`. Therefore the disableTypes logic with the ts|tsx replacement in the `generatedRoutTree`-File did not work.

This PR fixes the issue.

best regards

### CODE REVIEW

File: packages/router-vite-plugin/src/index.ts [hunk 1]
```diff
@@ -3,29 +3,16 @@ import { join, normalize } from 'path'
...
-  }
-}
-
-async function buildConfig(config: UserConfig, root: string): Promise<Config> {
-  const fileConfig = await readConfigFile(join(root, CONFIG_FILE_NAME))
-  return configSchema.parse({
-    ...fileConfig,
-    ...config,
-  })
+async function buildConfig(inlineConfig: UserConfig): Promise<Config> {
```
Comment (schiller-manuel) [inline packages/router-vite-plugin/src/index.ts:14]: can't we just call `getConfig` down there and remove this function?

Comment (wuarmin) [reply packages/router-vite-plugin/src/index.ts]: you are right, of course

File: packages/router-generator/src/config.ts [hunk 2]
```diff
@@ -15,17 +15,18 @@ export type Config = z.infer<typeof configSchema>
 
 const configFilePathJson = path.resolve(process.cwd(), 'tsr.config.json')
 
-export async function getConfig(): Promise<Config> {
+export async function getConfig(inlineConfig: Partial<Config> = {}): Promise<Config> {
```
Comment (schiller-manuel) [inline packages/router-generator/src/config.ts:18]: please pass in the directory as an optional argument here

```tsx
export async function getConfig(inlineConfig: Partial<Config> = {}, configDirectory?:string): Promise<Config> {
if (configDirectory === undefined) {
    configDirectory = process.cwd();
}
const configFilePathJson = path.resolve(configDirectory, 'tsr.config.json')

...

File: packages/router-vite-plugin/src/index.ts [hunk 3]
```diff
@@ -45,13 +26,13 @@ export function TanStackRouterVite(inlineConfig: UserConfig = {}): Plugin {
     name: 'vite-plugin-tanstack-router',
     configResolved: async (vite) => {
       ROOT = vite.root
-      userConfig = await buildConfig(inlineConfig, ROOT)
+      userConfig = await getConfig(inlineConfig)
```
Comment (schiller-manuel) [inline packages/router-vite-plugin/src/index.ts:29]: then pass in `ROOT` here

File: packages/router-vite-plugin/src/index.ts [hunk 4]
```diff
@@ -45,13 +26,13 @@ export function TanStackRouterVite(inlineConfig: UserConfig = {}): Plugin {
...
       await generate()
     },
     handleHotUpdate: async ({ file }) => {
       const filePath = normalize(file)
       if (filePath === join(ROOT, CONFIG_FILE_NAME)) {
-        userConfig = await buildConfig(inlineConfig, ROOT)
+        userConfig = await getConfig(inlineConfig)
```
Comment (schiller-manuel) [inline packages/router-vite-plugin/src/index.ts:35]: and here

**Review by schiller-manuel — CHANGES_REQUESTED** [review]

**Pull request merged**